src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/germline/CoverageModellerGermlineSparkToggle.java

```diff
@@ -0,0 +1,363 @@
+import org.broadinstitute.hellbender.tools.coveragemodel.CoverageModelParameters;
+import org.broadinstitute.hellbender.tools.coveragemodel.PosteriorVerbosityLevel;
+import org.broadinstitute.hellbender.tools.exome.*;
+import org.broadinstitute.hellbender.tools.exome.germlinehmm.IntegerCopyNumberState;
+import org.broadinstitute.hellbender.tools.exome.germlinehmm.IntegerCopyNumberTransitionProbabilityCacheCollection;
+import org.broadinstitute.hellbender.tools.exome.sexgenotyper.ContigGermlinePloidyAnnotationTableReader;
+import org.broadinstitute.hellbender.tools.exome.sexgenotyper.GermlinePloidyAnnotatedTargetCollection;
+import org.broadinstitute.hellbender.tools.exome.sexgenotyper.SexGenotypeDataCollection;
+import org.broadinstitute.hellbender.utils.SparkToggleCommandLineProgram;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.*;
+import java.util.Map;
+
+/**
+ * The command line tool for modelling and denoising coverage profiles and detecting germline copy number variation
+ *
+ * The tool can be run in two modes by specifying the job type (--jobType):
+ *
+ * <dl>
+ *     <dt>LEARN_AND_CALL</dt>
+ *     <dd>Learns model parameters from coverage profiles, saves the model, as well as various
+ *     posteriors (copy number, sample-specific biases, read depth, model-compatibility log likelihoods, etc.)
+ *     to disk</dd>
+ *
+ *     <dt>CALL_ONLY</dt>
+ *     <dd>Takes a previously learned model from the argument --inputModelPath, and calculates various posteriors
+ *     and saves the posteriors to disk</dd>
+ * </dl>
+ *
+ * The tool requires the following mandatory arguments:
+ *
+ * <dl>
+ *     <dt>Prior transition probabilities between different copy number states (specified via argument
+ *     --copyNumberTransitionPriorTable)</dt>
+ *
+ *     <dt>Combined raw or GC corrected (but not proportional) read counts table (specified via argument
+ *     --input)</dt>
+ *
+ *     <dt>Sample sex genotypes table (specified via argument --sexGenotypeTable)</dt>
+ *
+ *     <dt>Germline contig ploidy annotations for all sex genotypes (specified via argument
+ *     --contigAnnotationsTable)</dt>
+ *
+ *     <dt>Job type (specified via argument --jobType)</dt>
+ * </dl>
+ *
+ * The tool will automatically utilize Spark clusters if available. Otherwise, it will run in
+ * the single-machine (local) model.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+@CommandLineProgramProperties(
+        summary = "Performs coverage profile modelling, denoising, and detecting germline copy number variation",
+        oneLineSummary = "Performs coverage profile modelling, denoising, and detecting germline copy number variation",
+        programGroup = CopyNumberProgramGroup.class
+)
+public final class CoverageModellerGermlineSparkToggle extends SparkToggleCommandLineProgram {
```

Comment:
Regarding the tool name -- I agree that `CoverageModellerGermline` is a _terrible_ name and it _has_ to be renamed to something reasonable before the beta release. We cannot use `gCNV` either because it starts with lowercase `g` and it's just too restrictive (we will have `sCNV` soon and things will get complicated).

Perhaps we need a generic CLI tool called something like `FancySchmancyCNVCaller` that can perform the following tasks in its idealized form:
- create PoN and make calls from normals
- create PoN and make calls from tumors (possible with iHMM)
- create PoN and make calls from mixed normals and tumors (possible with iHMM)
- make calls from a given model on normals
- make calls from a given model on tumors
- make calls from a given model on mixed normals and tumors

The tool would then additionally take a sample annotation table (normal, tumor) and perform its job. For the first release, all samples have be annotated as normal; otherwise, an UnsupportedFeatureException is thrown.